src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -13820,6 +13821,16 @@ public static void RunModuleConstructor(System.ModuleHandle module) { }
         public static bool TryEnsureSufficientExecutionStack() { throw null; }
         public delegate void CleanupCode(object? userData, bool exceptionThrown);
         public delegate void TryCode(object? userData);
+        public static void UnsafeAwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : ICriticalNotifyCompletion { }
```

Comment:
I think we will want the runtime to build with async v2 feature disabled by default for .NET 10. Should the public surface changes and any changes in the default functionality be undef ifdef that's enabled only in the special build of the runtime with async v2 enabled?

(Also, any public surface changes will need to go through an API review before they show up in main.)

Reply:
> I think we will want the runtime to build with async v2 feature disabled by default for .NET 10.

Last I heard it was decided that we wanted to have the feature present, but codegen that targets it on C# side is hidden behind experimental flags, rhus requiring an opt-in. (i.e. virtual statics model?)
The presence of public API is a concern though. 

Also there are several ways to "turn off" the feature with varying effects on maintenance and testability. 

I think we should take this offline.